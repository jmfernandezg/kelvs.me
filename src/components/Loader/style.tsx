import styled from "styled-components";

export const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${props => props.theme.bg.light};
  color: ${props => props.theme.text.dark};
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99;

  .dark-mode & {
    background-color: ${props => props.theme.bg.dark};
    color: ${props => props.theme.text.light};
  }
`;

export const LogoWrapper = styled.div<{ isMounted: boolean }>`
  width: max-content;
  transition: all 0.25s cubic-bezier(0.645, 0.045, 0.355, 1);
  opacity: ${props => (props.isMounted ? 1 : 0)};
  svg {
    width: 100%;
    height: 100%;
    display: block;
    margin: 0 auto;
  }
`;
