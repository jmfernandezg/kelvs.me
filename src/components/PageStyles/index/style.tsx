import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: calc(100vh - 7rem);
  margin-top: 3em;
  ${({ theme }) => theme.desktop`
    justify-content: center;
    margin-bottom: 0;
    margin-top: 0;
    min-height: calc(65vh - 7rem);
    `}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 0 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.bigTablet`
    margin: 2em 0;
    svg {
      height: 1.1em;
    }
  `}
  ${({ theme }) => theme.thone`
    width: 40%;
  `}
`;

export const FeaturedProjects = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
`;

export const AppBox = styled.div`
  background: #fff 0 0 no-repeat padding-box;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  width: 32%;
  padding: 0.75em;
  margin-bottom: 1em;
  position: relative;
  transition: transform 0.2s;
  min-height: 290px;
  &:hover {
    transform: scale(1.025);
  }
  color: ${props => props.theme.text.dark};
  img {
    border: 1px solid #000;
    width: 100%;
    height: auto;
  }
  span {
    margin-right: 3px;
    background-color: #ff4e4e;
    color: #fff;
    padding: 0 3px 1px 3px;
    line-height: normal;
    font-size: 12px;
    font-weight: 300;
    letter-spacing: 1px;
    border-radius: 3px;
  }
  p {
    margin-bottom: 0;
  }
  ${({ theme }) => theme.tablet`
    width: 48%;
    p {
      font-size: 0.9rem;
    }
  `}
  ${({ theme }) => theme.thone`width: 100%;`}
`;

export const H1 = styled.h1`
  margin: 2.5em 0;
  ${({ theme }) => theme.tablet`
    margin: 2em 0;
    font-size: 1.125rem;
  `}
  font-weight: 700;
`;
