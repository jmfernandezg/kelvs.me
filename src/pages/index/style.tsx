import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  min-height: 40vh;
  ${({ theme }) => theme.tablet`justify-content: center;`}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 2em 0 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 4em 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.tablet`
    width: 40%;
    svg {
      height: 1.1em;
    }
  `}
`;
