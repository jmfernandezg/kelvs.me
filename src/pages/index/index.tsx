/* eslint-disable react/jsx-one-expression-per-line */
import Link from "next/link";
import Layout from "~/components/Layout";
import GmailSVG from "~/components/Icons/Gmail";
import GithubSVG from "~/components/Icons/Github";
import LinkedinSVG from "~/components/Icons/LinkedIn";
import { Container, HorizontalBar } from "~/components/PageStyles/common/style";
import {
  HeroWrapper,
  HeroOne,
  HeroTwo,
  CallToAction,
  Contact,
} from "~/components/PageStyles/index/style";

// https://www.justinewargnier.com/
// https://simpleicons.org/
// TODO: animate full stack dev text switch using react springs
export default function Home() {
  return (
    <Layout>
      <Container>
        <HeroWrapper>
          <HeroOne>
            <h1>
              Hello,&nbsp;
              <span>my name is Kelvin.</span>
            </h1>
            <HorizontalBar />
          </HeroOne>
          <HeroTwo>
            <h1>
              I&apos;m a <span>full&#8209;stack developer</span>
              &nbsp;and an open source enthusiast.
            </h1>
          </HeroTwo>
        </HeroWrapper>
        <CallToAction>
          <Link href="/about">
            <a>Learn more about me</a>
          </Link>
        </CallToAction>
        <Contact>
          <a href="https://github.com/kelvinabella">
            <GithubSVG />
          </a>
          <a href="https://www.linkedin.com/in/kelvinabella">
            <LinkedinSVG />
          </a>
          <a href="mailto:abellakelvin.ka@gmail.com">
            <GmailSVG />
          </a>
        </Contact>
      </Container>
    </Layout>
  );
}
