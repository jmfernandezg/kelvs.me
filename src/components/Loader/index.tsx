import { useState, useEffect } from "react";
import LoaderSVG from "~/components/Icons/Loader";
import { useScrollDisabler, animate } from "~/helpers/index";
import { LoaderContainer, LogoWrapper } from "./style";

type Props = { hideLoader: () => void };

export default function Loader({ hideLoader }: Props) {
  const [isMounted, setIsMounted] = useState(false);

  useScrollDisabler(true);

  useEffect(() => {
    // To avoid flickering of SVG.
    const timeout = setTimeout(() => setIsMounted(true), 10);
    animate(
      () => null,
      hideLoader,
      [
        { target: ".loader rect", offset: 0 },
        { target: ".loader path", offset: 400 },
      ],
      false,
      1500
    );
    return () => {
      clearTimeout(timeout);
    };
  }, [hideLoader]);

  return (
    <LoaderContainer>
      <LogoWrapper isMounted={isMounted}>
        <LoaderSVG />
      </LogoWrapper>
    </LoaderContainer>
  );
}
