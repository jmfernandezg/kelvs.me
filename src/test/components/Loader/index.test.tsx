import React from "react";
import { render } from "~/test/utils/index";
import Loader from "~/components/Loader";

// https://medium.com/@rickhanlonii/understanding-jest-mocks-f0046c68e53c
describe("Loader component", () => {
  it("renders Loader correctly", () => {
    window.scrollTo = jest.fn();
    const loaderMock = jest.fn();
    const { asFragment } = render(<Loader hideLoader={loaderMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
