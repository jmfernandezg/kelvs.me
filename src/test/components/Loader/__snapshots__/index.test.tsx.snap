// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders Loader correctly 1`] = `
<DocumentFragment>
  <div
    class="style__LoaderContainer-pts5p7-0 hlHeRz"
  >
    <div
      class="style__LogoWrapper-pts5p7-1 bUjgMh"
    >
      <svg
        class="loader"
        fill="none"
        height="90"
        viewBox="0 0 317 90"
        width="317"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M52.7385 22.1H62.3815L38.3491 41.5876L37.9602 41.903L38.276 42.2914L62.0212 71.5H53.0883L33.0784 46.4877L32.7648 46.0956L32.3741 46.4108L19.3421 56.9228L19.156 57.0729V57.312V71.5H12.38V22.1H19.156V48.384V49.4299L19.97 48.7731L32.642 38.5492L32.6447 38.5469L52.7385 22.1ZM91.8669 65.012H123.407V71.5H84.5909V22.1H123.407V28.588H91.8669H91.3669V29.088V42.408V42.908H91.8669H119.447V49.396H91.8669H91.3669V49.896V64.512V65.012H91.8669ZM153.883 65.012H185.063V71.5H146.607V22.1H153.383V64.512V65.012H153.883ZM217.76 71.5L195.53 22.1H203.285L216.328 51.962C216.328 51.9631 216.329 51.9643 216.329 51.9654C217.099 53.7939 217.989 55.8134 218.998 58.0237L219.002 58.0317C220.052 60.229 220.959 62.3284 221.721 64.33L222.175 65.5195L222.652 64.3396C223.467 62.3274 224.329 60.243 225.24 58.0866C226.201 55.9231 227.043 53.9017 227.766 52.0227C227.766 52.0217 227.767 52.0206 227.767 52.0195L239.806 22.1H247.846L226.11 71.5H217.76ZM266.75 57.2923C269.306 60.0468 272.023 62.2182 274.905 63.7985L274.91 63.8014C277.974 65.432 281.635 66.236 285.873 66.236C288.219 66.236 290.495 65.9427 292.698 65.3551C294.912 64.7647 296.737 63.8715 298.151 62.6596L298.152 62.6587C299.653 61.3646 300.413 59.7147 300.413 57.744C300.413 56.0068 299.718 54.6079 298.34 53.5992C297.056 52.6247 295.2 51.9097 292.818 51.4231C290.448 50.8913 287.603 50.4095 284.284 49.9765C280.235 49.4047 276.723 48.549 273.744 47.4144C270.829 46.2385 268.603 44.6992 267.036 42.8116C265.488 40.8978 264.701 38.5184 264.701 35.64C264.701 32.6319 265.623 30.1131 267.455 28.0522L267.458 28.0481C269.314 25.9148 271.805 24.2787 274.954 23.1507L274.958 23.1493C278.16 21.9721 281.7 21.38 285.585 21.38C290.43 21.38 294.584 22.1402 298.059 23.6429L298.061 23.6439C301.446 25.0882 304.098 27.1085 306.035 29.6988L301.518 34.5928C299.758 32.5299 297.594 30.8353 295.031 29.508C292.26 28.0729 289.034 27.364 285.369 27.364C281.669 27.364 278.539 27.9962 276.002 29.2884C274.726 29.9147 273.735 30.6947 273.063 31.643C272.387 32.5971 272.053 33.6956 272.053 34.92C272.053 36.7786 272.706 38.3121 274.023 39.4715L274.032 39.4789C275.306 40.5497 277.062 41.3677 279.264 41.9551C281.447 42.5371 283.986 43.0445 286.879 43.4785L286.886 43.4795L286.893 43.4804C289.662 43.8145 292.309 44.2677 294.835 44.8397C297.348 45.4087 299.566 46.1652 301.494 47.1054L301.498 47.1074C303.446 48.035 304.96 49.2784 306.057 50.8323L306.062 50.8392L306.067 50.8459C307.187 52.3245 307.765 54.2272 307.765 56.592C307.765 59.7532 306.861 62.5008 305.06 64.8563L305.059 64.8575C303.301 67.1702 300.768 68.9928 297.427 70.3104C294.138 71.5784 290.219 72.22 285.657 72.22C280.765 72.22 276.363 71.4602 272.445 69.9495C268.661 68.4907 265.291 66.1466 262.334 62.9062L266.75 57.2923Z"
          stroke="currentColor"
          stroke-dasharray="undefined"
          style="stroke-dashoffset: 0; stroke-dashoffset: 0;"
        />
        <rect
          height="89"
          rx="4.5"
          stroke="currentColor"
          stroke-dasharray="810"
          style="stroke-dashoffset: 810;"
          width="316"
          x="0.5"
          y="0.5"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
